handsontable/src/selection/utils.js

```diff
@@ -222,3 +223,40 @@ export function transformSelectionToRowDistance(selectionRanges) {
 
   return normalizedRowRanges;
 }
+
+/**
+ * Returns an object with `rows` and `columns` keys. The arrays contains sorted indexes
+ * generated according to the given `ranges` array.
+ *
+ * @param {Array<{startRow: number, startCol: number, endRow: number, endCol: number}>} ranges The range to process.
+ * @returns {{rows: number[], columns: number[]}}
+ */
+export function normalizeRanges(ranges) {
```

Comment:
As you moved the function from the CopyPaste context to Selection, I'd rename the function to a more meaningful name. I'm thinking about something like `transformRangeLikeToRecordsList` or `transformRangeLikeToRecords` - or something other, as it accepts not the `CellRange` instances but literal objects with `startRow`, `startCol` ... props.